manifests/init.pp

```diff
@@ -76,53 +76,10 @@
   }
 
   if $manage_repo {
-    case $facts['os']['family'] {
-      'Debian': {
-
-        apt::source { 'apt_gitlabci':
-          comment  => 'GitlabCI Runner Repo',
-          location => "${repo_base_url}/runner/${package_name}/${facts['os']['distro']['id'].downcase}/",
-          repos    => 'main',
-          key      => {
-            'id'     => '1A4C919DB987D435939638B914219A96E15E78F4',
-            'server' => $repo_keyserver,
-          },
-          include  => {
-            'src' => false,
-            'deb' => true,
-          },
-        }
-        Apt::Source['apt_gitlabci'] -> Package[$package_name]
-        Exec['apt_update'] -> Package[$package_name]
-      }
-      'RedHat': {
-        yumrepo { "runner_${package_name}":
-          ensure        => 'present',
-          baseurl       => "${repo_base_url}/runner/${package_name}/el/\$releasever/\$basearch",
-          descr         => "runner_${package_name}",
-          enabled       => '1',
-          gpgcheck      => '0',
-          gpgkey        => "${repo_base_url}/gpg.key",
-          repo_gpgcheck => '1',
-          sslcacert     => '/etc/pki/tls/certs/ca-bundle.crt',
-          sslverify     => '1',
-        }
-
-        yumrepo { "runner_${package_name}-source":
-          ensure        => 'present',
-          baseurl       => "${repo_base_url}/runner/${package_name}/el/\$releasever/SRPMS",
-          descr         => "runner_${package_name}-source",
-          enabled       => '1',
-          gpgcheck      => '0',
-          gpgkey        => "${repo_base_url}/gpg.key",
-          repo_gpgcheck => '1',
-          sslcacert     => '/etc/pki/tls/certs/ca-bundle.crt',
-          sslverify     => '1',
-        }
-      }
-      default: {
-        fail ("gitlab_ci_runner::manage_repo parameter for ${facts['os']['family']} is not supported.")
-      }
+    class { 'gitlab_ci_runner::repo':
```

Comment:
could you please update the class so it has default values? That allows us to do a simple `contain gitlab_ci_runner::repo` here. (Which is the current best practice).

Reply:
Hey @bastelfreak, could you please point me to the relevant best practice documentation? I'm actually not aware of this best practice.